merlin/core/dispatch.py

```diff
@@ -293,11 +293,18 @@ def list_val_dtype(ser: SeriesLike) -> np.dtype:
             return pd.core.dtypes.cast.infer_dtype_from(next(iter(pd.core.common.flatten(ser))))[0]
     if isinstance(ser, np.ndarray):
         return ser.dtype
+    # adds detection when in merlin column
+    if hasattr(ser, "is_list"):
+        return ser[0].dtype
     return None
 
 
 def is_list_dtype(ser):
     """Check if Series contains list elements"""
+    # adds detection for merlin column
+    if hasattr(ser, "is_list"):
+        if isinstance(ser.values[0], np.ndarray):
```

Comment:
Unfortunately the problem this fixes for now, while we are in flux is that dtype for an np nd array is object because of the scenario with the list of one item. In that case is_list comes back as not true. because your shape in the case that you have the following [[1],[2],[3]] the shape of this is (1,) which is the same as a scalar so it doesnt meet the requirements yet. However once merlin dtypes are in place and Columns use those dtypes we can swap out this logic. But for now, we have to keep it like this.

Reply:
Could this logic live inside the implementation of `is_list` in the Column class? even without the dtype integration, this seems like something we could handle already in there?

Reply:
It could but I dont believe it should. And again this is to allow for local runtime to work given current circumstances. I think the job of detecting whether its a list or not belongs to the dtype. Which is where it lives in the other frameworks... i.e. cudf- https://docs.rapids.ai/api/cudf/nightly/user_guide/cudf.ListDtype.html#cudf.ListDtype and pandas  https://pandas.pydata.org/pandas-docs/stable/user_guide/basics.html#basics-dtypes) This something we are thinking about delegating to the merlin dtype to follow that same pattern. It has not lived there in the past because we didnt have merlin dtypes.

Reply:
I went ahead and made the change.